### DISCUSSION TITLE
Partial path completions (aka fuzzy completions)

### DISCUSSION THREAD

**Discussion by yerlaser** [nushell/nushell#5895] [Q&A] [answered]
In Release Notes for 0.63 it was mentioned partial path completions should work.
I assumed that this would be:
`cd p/t/f` `TAB`
to allow completion for:
`cd path/to/folder`
but it doesn't work.
Do I need to turn it on somehow?

**Comment by fdncred** [comment]
If you do `cd p/t/f<enter>` it will cd to path/to/folder. This abbreviated folder thing isn't a completion yet.

**Reply by yerlaser** [reply]
Just tried on 0.63.1 (compiled from sources with --features=extra) option.
Doesn't work :(

**Reply by yerlaser** [reply]
Here's the error message:
Error: nu::shell::directory_not_found (link)

  × Directory not found
   ╭─[entry #41:1:1]
 1 │ cd to/je
   ·    ──┬──
   ·      ╰── directory not found
   ╰────
  help: IO Error: Os { code: 2, kind: NotFound, message: "No such file or directory" }


There's this folder structure:
`todo/jefferson`

and that's the only combination that fits `to/je`

**Comment by fdncred** [comment, marked as answer]
It's working on version 0.64
```
>cd n/p 
before: [/Users/fdncred/src/forks] after: [/Users/fdncred/src/forks/nushell/pkg_mgrs]
>cd - 
before: [/Users/fdncred/src/forks/nushell/pkg_mgrs] after: [/Users/fdncred/src/forks]
>cd n/pr
before: [/Users/fdncred/src/forks] after: [/Users/fdncred/src/forks/nu_scripts/progress_bar]
>cd -
before: [/Users/fdncred/src/forks/nu_scripts/progress_bar] after: [/Users/fdncred/src/forks]
>cd n/prom
before: [/Users/fdncred/src/forks] after: [/Users/fdncred/src/forks/nu_scripts/prompt]
```
note that there is a configuration setting to enable this functionality too.
https://github.com/nushell/nushell/pull/5452

**Reply by yerlaser** [reply]
Ah, yes, that config I missed.
Thanks for the answer!

P.S. For those who come across the same issue you need to add:
`cd_with_abbreviations: true`
into your config.nu

**Comment by lavafroth** [comment]
Tested this on version 0.80.0, we now only need to change `abbreviations` from `false` to `true` under `cd` in the config file.

```nu
let-env config = {
    # ...
    cd: {
        abbreviations: true
    }
}
```

Thanks for the feature!
